library/core/src/pin.rs

```diff
@@ -802,6 +802,44 @@ impl<T: ?Sized> Pin<&'static T> {
     }
 }
 
+impl<'a, P: DerefMut> Pin<&'a mut Pin<P>> {
+    /// Gets a pinned mutable reference from this nested pinned pointer.
+    ///
+    /// This is a generic method to go from `Pin<&mut Pin<Pointer<T>>>` to `Pin<&mut T>`. It is
+    /// safe because the existence of a `Pin<Pointer<T>>` ensures that the pointee, `T`, cannot
+    /// move in the future, and this method does not enable the pointee to move. "Malicious"
+    /// implementations of `Pointer::DerefMut` are likewise ruled out by the contract of
```

Comment:
```suggestion
    /// implementations of `P::DerefMut` are likewise ruled out by the contract of
```

Reply:
I'm not sure I follow this suggestion — this is specifically referring to the `Pointer` type mentioned earlier in the text. I suppose in this case `P = Pointer`, but is that clearer in context?

Reply:
I was trying to match this up with the actual type of the method, and there is no `Pointer` there -- so that seems confusing. `self` has type `Pin<&'a mut Pin<P>>`.